android/fakedapp/src/main/java/com/solana/mobilewalletadapter/fakedapp/MainViewModel.kt

```diff
@@ -419,56 +373,35 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
         action: suspend (MobileWalletAdapterClient) -> T?
     ): T? {
         return mobileWalletAdapterClientMutex.withLock {
-            val semConnectedOrFailed = Semaphore(1, 1)
-            val semTerminated = Semaphore(1, 1)
-            var mobileWalletAdapterClient: MobileWalletAdapterClient? = null
-            val scenarioCallbacks = object : Scenario.Callbacks {
-                override fun onScenarioReady(client: MobileWalletAdapterClient) {
-                    mobileWalletAdapterClient = client
-                    semConnectedOrFailed.release()
-                }
-
-                override fun onScenarioError() = semConnectedOrFailed.release()
-                override fun onScenarioComplete() = semConnectedOrFailed.release()
-                override fun onScenarioTeardownComplete() = semTerminated.release()
-            }
-
             val localAssociation = LocalAssociationScenario(
                 getApplication<Application>().mainLooper,
                 Scenario.DEFAULT_CLIENT_TIMEOUT_MS,
-                scenarioCallbacks,
                 uriPrefix
             )
             sender.startActivityForResult(localAssociation.createAssociationIntent())
 
-            localAssociation.start()
-            try {
-                withTimeout(ASSOCIATION_TIMEOUT_MS) {
-                    semConnectedOrFailed.acquire()
+            return@withLock withContext(Dispatchers.IO) {
```

Comment:
Again, I think a reasonable outcome of this PR is removing the need to have the mutex at all.